monitoring/monitoring.yaml

```diff
@@ -416,3 +416,255 @@ spec:
      targetPort: 3000
   selector:
     app: grafana
+---
+kind: Service
+apiVersion: v1
+metadata:
+  name: elasticsearch
+  namespace: monitoring
+  labels:
+    app: elasticsearch
+spec:
+  selector:
+      app: elasticsearch
+  clusterIP: None
+  ports:
+    - port: 9200
+      name: rest
+    - port: 9300
+      name: inter-node
+---
+apiVersion: apps/v1
+kind: StatefulSet
+metadata:
+  name: es-cluster
+  namespace: monitoring
+spec:
+  serviceName: elasticsearch
+  replicas: 3
+  selector:
+    matchLabels:
+      app: elasticsearch
+  template:
+    metadata:
+      labels:
+        app: elasticsearch
+    spec:
+      containers:
+        - name: elasticsearch
+          image: docker.elastic.co/elasticsearch/elasticsearch-oss:7.1.1
+          resources:
+            limits:
+              cpu: 1000m
+            requests:
+              cpu: 100m
+          ports:
+            - containerPort: 9200
+              name: rest
+              protocol: TCP
+            - containerPort: 9300
+              name: inter-node
+              protocol: TCP
+          volumeMounts:
+            - name: data
+              mountPath: /usr/share/elasticsearch/data
+          env:
+            - name: cluster.name # Should this name be hardcoded?
+              value: gke_hail-vdc_us-central1-a_vdc
+            - name: node.name
+              valueFrom:
+                fieldRef:
+                  fieldPath: metadata.name
+            - name: discovery.zen.ping.unicast.hosts
+              value: "es-cluster-0.elasticsearch,es-cluster-1.elasticsearch,es-cluster-2.elasticsearch" #TODO This seems dumb, why must I list out all numbers?
+            - name: discovery.zen.minimum_master_nodes
+              value: "2" #Should be N/2 + 1 at minimum, where N is the number of replicas.
```

Comment:
again, with Jinja you could do 

```
{% set n_replicas = 3 %}
...
replicas: {{ n_replicas }}
...
          value: {{ n_replicas // 2 + 1 }}
```

Though I like the parameter to be failure_tolerance and set replicas to failure_tolerance * 2 + 1